samples/kotlin-multiplatform-js/build.gradle.kts

```diff
@@ -0,0 +1,16 @@
+import org.jlleitschuh.gradle.ktlint.KtlintExtension
+
+apply {
+    plugins.apply("org.jlleitschuh.gradle.ktlint")
```

Comment:
Uhhh... This doesn't seem right.

Reply:
This is how all samples are implemented, check this line in main `build.gradle.kts`: https://github.com/JLLeitschuh/ktlint-gradle/blob/27b5e6586c1b1973ba19209b913ea53a3ff583cd/build.gradle.kts#L8

Reply:
So each sample module apply plugin from included plugin project using `plugins.apply()`

Reply:
I think that this is what you want instead.
```kotlin
plugins {
    // This is a better API as the Kotlin DSL can automatically generate accessors for the `KtlintExtension` 
    id("org.jlleitschuh.gradle.ktlint")
}
apply {
    plugin("kotlin-platform-js")
}
```

If you use the above syntax, you should get this for free:
```kotlin
ktlint {
    // Your configuration goes here.
}
```

Let me know if this isn't the case.

Reply:
But will this:

Reply:
Sorry, maybe I was not clear. 

Current approach uses compiled plugin from plugin project included build: https://github.com/JLLeitschuh/ktlint-gradle/blob/27b5e6586c1b1973ba19209b913ea53a3ff583cd/settings.gradle.kts#L40

So when someone changes something in plugin project samples will compile it and apply to their configuration.

By using `plugins {}` block samples will always try to download latest plugin version from some artifactory and I can't find a way to tell it to use included build.

Also I think this part of Gradle documentation relates to this particular use-case: https://docs.gradle.org/4.5.1/userguide/plugins.html#sec:applying_plugins_buildscript
And it advises to use current approach.

Reply:
Ah, I see. Okay.
This is related to this issue:
https://github.com/gradle/gradle/issues/2528

Okay, fair enough.
Regardless, `plugins` doesn't need to be inside of the `apply` as the receiver of the `plugins` call is on `Project`.
So you can do this instead:
```kotlin
apply {
    plugin("kotlin-platform-js")
}
plugins.apply("org.jlleitschuh.gradle.ktlint")
```